### ISSUE TITLE
What is the best/effiecient approach for utilizing and working on a text dataset?

### ISSUE THREAD

**Issue by ankitvad** [PyTorchLightning/pytorch-lightning#5770]
Hi,
So, I've been researching different approaches/libraries online that work for text dataset - torchtext, PyTorch-NLP, pytext, AllenNLP but most of them are designed to work individualy or on a single machine. 

If I am working on text dataset for tasks like seq2seq; what is the most efficient data sampling/generation approach that can utilize all PytorchLightning features? For example: parallel training and 16-bit apex precision. 

Ideally what I'd like to do is be able to - create batches, vocabulary, somehow sort the dataset and batch them in such a way that I don't need to truncate the text item and don't need to do a lot of padding like torchtext does with it's BucketIterator. 

I've had a talk with some community members in the past where I was offered smart ways to use torchtext library in Lightning but I'd like to reduce my dependency on that legacy code until it's essential. For eg: torchtext is phasing out most of its iterators and Field and Dataset classes.  

So it really messes stuff up when all other libraries update but the old code is just made legacy.

**Comment by carmocca** [comment]
Hi! Lightning does not provide any custom batch samplers or other abstractions to solve the specific issues of variable length data. However, you can use any of those you mentioned in a LightningDataModule for your particular needs.

> what is the most efficient data sampling/generation approach that can utilize all PytorchLightning features?

This is too broad of a question to have an answer, but in general, the most efficient approach for PyTorch will also be the most efficient approach for Lightning.

**Comment by ankitvad** [comment]
That makes sense! Thanks!
